megatron/mpu/mappings.py

```diff
@@ -136,6 +188,53 @@ def forward(ctx, input_):
     def backward(ctx, grad_output):
         return _split(grad_output)
 
+class _ReduceScatterFromModelParallelRegion(torch.autograd.Function):
```

Comment:
@siddharth9820 -- where is this ReduceScatter being used?

Reply:
@awan-10, for now nowhere. I can remove it if you want. I kept it there because it can potentially be used in place of the all-reduce at the end of tensor-parallel self-attention.

Reply:
Got it. Good to remove from this PR but don't change things just yet. All of this code essentially should be inside deepspeed and not here. I can share more with you when we talk.